envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/listeners/filters/RBACFilterPermissions.kt

```diff
@@ -44,18 +48,45 @@ class RBACFilterPermissions {
     }
 
     private fun createPathMatcher(path: String, matchingType: PathMatchingType): PathMatcher {
-        val matcher = when (matchingType) {
-            PathMatchingType.PATH -> StringMatcher.newBuilder().setExact(path).build()
-            PathMatchingType.PATH_PREFIX -> StringMatcher.newBuilder().setPrefix(path).build()
+        val matcher = when (path.contains(paramRegex)) {
+            true -> {
+                val regexPath = getRegexPath(path, matchingType)
+                StringMatcher.newBuilder()
+                    .setSafeRegex(
+                        RegexMatcher.newBuilder()
+                            .setRegex(regexPath)
+                            .setGoogleRe2(
+                                RegexMatcher.GoogleRE2.getDefaultInstance()
+                            )
+                            .build()
+                    )
+                    .build()
+            }
+            false -> when (matchingType) {
+                PathMatchingType.PATH -> StringMatcher.newBuilder().setExact(path).build()
+                PathMatchingType.PATH_PREFIX -> StringMatcher.newBuilder().setPrefix(path).build()
+            }
         }
         return PathMatcher.newBuilder().setPath(matcher).build()
     }
+
+    private fun getRegexPath(path: String, matchingType: PathMatchingType): String {
+        var regexPath = path.replace(paramRegex, "\\\\w+")
+        regexPath = regexPath.replace("/", "\\/")
+        if (matchingType == PathMatchingType.PATH_PREFIX) {
+            regexPath += ".*"
+        } else if (matchingType == PathMatchingType.PATH && regexPath.endsWith(".*")) {
```

Comment:
What if I add `.*` in any place of path?

Reply:
All `.*` are removed. Just adding `.*` at the end in the case of pathPrefix type